conda → miniforge. Make anaconda explicitly red. (#444)
UCL-ARC/python-tooling: docs/pages/virtual.md

| [pyenv](https://github.com/pyenv/pyenv)                           | Lets you easily switch between multiple versions of Python.                                                                                                                               | 🟠  |
| [virtualenv](https://virtualenv.pypa.io/en/latest/)               | Creates isolated Python environments, and offers more features than venv.                                                                                                                 | 🟠  |
| [venv](https://docs.python.org/3/library/venv.html)               | Creates isolated Python environments.                                                                                                                                                     | 🔴  |
| [anaconda](https://www.anaconda.com/)                             | Due to recent [licensing ambiguity][anaconda-problems], we recommend avoiding anaconda and many of the default channels. We recommend miniforge or sticking to the `conda-forge` channel. | 🔴  |

> [anaconda-problems]: https://www.theregister.com/2024/08/08/anaconda_puts_the_squeeze_on/

Reviewer: I know I've said this before, I really dislike this syntax. But, I hate inconsistency even more.

Author: But then the table becomes huge.

Reviewer: But it's fine because `prettier` just does its thing?

Author: Maybe I'm just weird in that I read the .md directly. 

Anyway: I pulled them all out. 😛

Reviewer: I know 😢 at least it's consistent. Even though the syntax hurts my eyes.

Reviewer: Especially if you "read the .md directly", it's like looking at footnotes all the time

Reviewer 2: FWIW I quite like this syntax, and agree it makes the table more readable in its raw markdwon.

Reviewer: https://google.github.io/styleguide/docguide/style.html#links

Reviewer 2: That style guide seems to [expcicitly recommend references for long links](https://google.github.io/styleguide/docguide/style.html#reference):

> For long links or image URLs, you may want to split the link use from the link definition

Reviewer: That was me admitting defeat, but I still won't use it myself

# Virtual environments

| Name                    | Short description                                                                                                                                                                         | 🚦  |
| ----------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | :-: |

> | conda-forge [miniforge] | Installs, runs, and updates packages and their dependencies. A trusted sub-set of `conda` with the `conda-forge` channel.                                                                 | 🟢  |

Reviewer 2: ```suggestion
| conda-forge [miniforge] | Installs, runs, and updates packages and their dependencies. Similar to `conda`, but uses community maintained packages from `conda-forge` channel instead of commercially maintained packages.                                                                 | 🟢  |
```
Everything on conda-forge is packaged independently to conda, so I think sub-set isn't the right word.

Author: Thanks for the comments! That is the specificity I was looking for.

I don't know where I got "subset" from. I think my brain morphed "minimal installers" into that somehow...

> [...] the minimal installers for [Conda](https://conda.io/) and [Mamba](https://github.com/mamba-org/mamba) specific to [conda-forge](https://conda-forge.org/), [...]

Author: It's not really "Similar to" though? It _is_  `conda`, right?

Author: How about
> Uses `conda` but with community managed...

Reviewer 2: Ah yes, ☝️ is much better

| [pipenv]                | Automatically creates and manages a virtualenv for your projects.                                                                                                                         | 🟠  |
| [pyenv]                 | Lets you easily switch between multiple versions of Python.                                                                                                                               | 🟠  |
| [virtualenv]            | Creates isolated Python environments, and offers more features than venv.                                                                                                                 | 🟠  |

> | [anaconda]              | Due to recent [licensing ambiguity][anaconda-problems], we recommend avoiding anaconda and many of the default channels. We recommend miniforge or sticking to the `conda-forge` channel. | 🔴  |

Reviewer 2: ```suggestion
| [anaconda]              | Due to recent [licensing ambiguity][anaconda-problems], we recommend avoiding anaconda and many of the default channels. We recommend installing miniforge and sticking to the `conda-forge` channel. | 🔴  |
```

| [venv]                  | Creates isolated Python environments.                                                                                                                                                     | 🔴  |

<!-- links here for a more readable table -->

> [miniforge]: https://github.com/conda-forge/miniforge

Reviewer 2: ```suggestion
[miniforge]: https://conda-forge.org/download/
```
Is maybe a slightly more user friendly link that gets you to an installer quicker?

[pipenv]: https://pipenv.pypa.io/en/latest/
[pyenv]: https://github.com/pyenv/pyenv
[virtualenv]: https://virtualenv.pypa.io/en/latest/
[anaconda]: https://www.anaconda.com/
[anaconda-problems]: https://www.theregister.com/2024/08/08/anaconda_puts_the_squeeze_on/

> [venv]: https://commons.wikimedia.org/wiki/File:Devops-toolchain.svg

Reviewer 3: probably that was changed by mistake
```suggestion
[venv]: https://docs.python.org/3/library/venv.html
```

Reviewer: Wouldn't have happened with inline...

Author: I'm _so_ good at computers.
